qtconsole/console_widget.py

```diff
@@ -396,6 +399,7 @@ def eventFilter(self, obj, event):
             text widgets.
         """
         etype = event.type()
+        self._trigger_is_complete_callback()
```

Comment:
The user may have previously pressed enter and we didn't know whether we should execute or not and thus asked the kernel if the command is complete. If the kernel was to slow to respond, we simply assume the command is not complete before we process any more user input.